### PR TITLE
[BugFix]fixed SQLBLACKLIST row contains multiple spaces and line separators

### PR DESCRIPTION
**Pull request by liubotao**
when SQLBLACKLIST sql contains multiple spaces and line separators, the forbidden SQL can pass 。Adding multiple conditions to  verifying sql 。

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
- [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by kangkaisen — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/meta/SqlBlackList.java [hunk 1]
```diff
@@ -35,8 +35,9 @@ public static SqlBlackList getInstance() {
     }
 
     public static void verifying(String sql) throws AnalysisException {
+        String formatSql = sql.replace("\r", " ").replace("\n", " ").replaceAll("\\s+", " ");
         for (BlackListSql patternAndId : getInstance().sqlBlackListMap.values()) {
-            Matcher m = patternAndId.pattern.matcher(sql);
+            Matcher m = patternAndId.pattern.matcher(formatSql);
```
Comment (packy92) [inline fe/fe-core/src/main/java/com/starrocks/meta/SqlBlackList.java:40]: We only remove space when building balcklist pattern. Should we also need to remove line sperator here?
```
public void analyze() throws SemanticException {
        sql = sql.trim().toLowerCase().replaceAll(" +", " ");
        if (sql.length() > 0) {
            try {
                sqlPattern = Pattern.compile(sql);
            } catch (PatternSyntaxException e) {
                throw new SemanticException("Sql syntax error: %s", e.getMessage());
            }
        }
    }
```

File: fe/fe-core/src/test/java/com/starrocks/planner/QueryPlannerTest.java [hunk 2]
```diff
@@ -165,6 +165,16 @@ public void testSqlBlackList() throws Exception {
...
+        Assert.assertEquals("Access denied; This sql is in blacklist, please contact your admin",
+                connectContext.getState().getErrorMessage());
+        connectContext.getState().setError("");
+
+        String sqlWithLineSeparators = "select k1 \n" +
+                " from \n" +
+                " test.baseall";
+        StatementBase statement1 = SqlParser.parse(sqlWithLineSeparators, connectContext.getSessionVariable().getSqlMode()).get(0);
+        StmtExecutor stmtExecutor4 = new StmtExecutor(connectContext, statement1);
+        stmtExecutor4.execute();
```
Comment (packy92) [inline fe/fe-core/src/test/java/com/starrocks/planner/QueryPlannerTest.java:177]: Could you add some sql-tester tests?

File: fe/fe-core/src/main/java/com/starrocks/meta/SqlBlackList.java (hunk 1 above)
Comment (liubotao) [reply fe/fe-core/src/main/java/com/starrocks/meta/SqlBlackList.java:40]: We need stricter syntax rule checking when the user adds the sql to the Backlist, so I don't think we need to remove the separator here.

File: fe/fe-core/src/test/java/com/starrocks/planner/QueryPlannerTest.java (hunk 2 above)
Comment (liubotao) [reply fe/fe-core/src/test/java/com/starrocks/planner/QueryPlannerTest.java:177]: testSqlBlackList  contains sqlWithNoLineSeparators and sqlWithLineSeparators Test Cases

File: fe/fe-core/src/main/java/com/starrocks/meta/SqlBlackList.java (hunk 1 above)
Comment (liubotao) [reply fe/fe-core/src/main/java/com/starrocks/meta/SqlBlackList.java:40]: We try not to modify user inputs in case they are confused

Comment (liubotao) [reply fe/fe-core/src/main/java/com/starrocks/meta/SqlBlackList.java:40]: Yes, Added

**Comment by packy92** [comment]
seems need a rebase?

**Pull request closed without merging**